### PR TITLE
Migrate to clang-format-16

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit upgrades clang-format to 16.

ONE-DCO-1.0-Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

---

Related issue: #12311

### CODE REVIEW

File: .github/workflows/check-format.yml [hunk 1]
```diff
@@ -28,11 +28,14 @@ jobs:
...
 
-      # C format: clang-format-8
+      # C format: clang-format-16
       # Python format: yapf==0.22.0
       - name: Install packages
         run: |
-          sudo apt-get install -y clang-format-8
+          sudo apt-get install -y gnupg2 software-properties-common
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+          sudo add-apt-repository "deb http://apt.llvm.org/focal/ llvm-toolchain-focal-16 main"
```
Comment (seanshpark) [inline .github/workflows/check-format.yml:37]: Q) if we use 22.04, use `jammy` not `focal`?

Comment (hseok-oh) [reply .github/workflows/check-format.yml:37]: It's mistake.

File: onert-micro/externals/flatbuffers/flexbuffers.h [hunk 2]
```diff
@@ -1595,11 +1595,10 @@ class Builder FLATBUFFERS_FINAL_CLASS
   template <typename T> static Type GetScalarType()
   {
     static_assert(flatbuffers::is_scalar<T>::value, "Unrelated types");
-    return flatbuffers::is_floating_point<T>::value
-             ? FBT_FLOAT
-             : flatbuffers::is_same<T, bool>::value
-                 ? FBT_BOOL
-                 : (flatbuffers::is_unsigned<T>::value ? FBT_UINT : FBT_INT);
+    return flatbuffers::is_floating_point<T>::value ? FBT_FLOAT
```
Comment (seanshpark) [inline onert-micro/externals/flatbuffers/flexbuffers.h:1598]: maybe we should exclude `externals`

Comment (hseok-oh) [reply onert-micro/externals/flatbuffers/flexbuffers.h:1598]: I'll add `.FORMATDENY` file into `onert-micro/externals` to ignore format checking.

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

NOTE: approval for changes of mixed modules in compiler is done as
- format change is machine translation from clang-format, not human jobs
- splitting compiler modules is labor intensive which is not needed for this case

**Pull request merged**